app/controllers/signupAuth.js

```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+
+const User = mongoose.model('User');
+const jwt = require('jsonwebtoken');
+const moment = require('moment');
+
+const secret = process.env.SECRET_TOKEN_KEY;
+
+
+module.exports.signup = (req, res) => {
+  const body = req.body;
+
+  if (!(body.name || body.email || body.password)) {
+    return res.json({ success: false,
+      message: 'Incomplete information. name, email and password are required.'
+    });
+  }
+  const newUser = new User({
+    name: req.body.name,
+    email: req.body.email,
+    password: req.body.password
+  });
+
+  newUser.save((err, user) => {
+    if (err) {
```

Comment:
use return to avoid the else e.g 

```js
if(err) {
  return res.json(err)
}

// do your 

return res.json({})
```
This makes you code tidy and easy to follow